packages/plugin-ember/src/hooks/register-tasks.ts

```diff
@@ -3,10 +3,10 @@ import { DependenciesTask, EmberProjectTask, TypesTask } from '../tasks';
 import { Hook } from '@oclif/config';
 
 // TODO: Determine correct type for options
-const hook: Hook<'register-tasks'> = async function({ registerTask }: any) {
-  registerTask(EmberProjectTask.taskName, EmberProjectTask);
-  registerTask(DependenciesTask.taskName, DependenciesTask);
-  registerTask(TypesTask.taskName, TypesTask);
+const hook: Hook<'register-tasks'> = async function({ cliArguments, tasks }: any) {
+  tasks.registerTask(EmberProjectTask.taskName, new EmberProjectTask(cliArguments));
+  tasks.registerTask(DependenciesTask.taskName, new DependenciesTask(cliArguments));
+  tasks.registerTask(TypesTask.taskName, new TypesTask(cliArguments));
```

Comment:
Was going to do this today, but looks like you could probably just do it as part of this refactor.

We can just pass in the instance and add `taskName` to the interface of `Task` so that all tasks must define one. This allows us to then have `bucket` and `bucketPriority` (or something similar) to `Task` so they can be referenced when we are sorting the run/output order.

```suggestion
  tasks.registerTask(new EmberProjectTask(cliArguments));
  tasks.registerTask(new DependenciesTask(cliArguments));
  tasks.registerTask(new TypesTask(cliArguments));
```

Reply:
Ya I thought of this, and this particular code is taking shape as I go along. One thing is that this value is constant - it doesn't really change for each instance (though it's a moot point, as we only ever have one instance). Making it static did seem correct, but I _do_ agree with the ability to enforce implementers to implement these properties (though we can't really enforce they've initialized them correctly except via validation, which could be done right here as tasks are added).

Let me finalize some of the things I'm doing right now and I'll move ahead with this next part.

Reply:
As per conversation, we will tackle this at a later point in time.